src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -86,10 +86,31 @@ public void validate(@Nullable Schema inputSchema) {
         }
 
         Schema fieldSchema = BigQueryUtil.getNonNullableSchema(field.getSchema());
+        Schema.Type type = fieldSchema.getType();
+        String name = field.getName();
 
-        if (!fieldSchema.getType().isSimpleType()) {
-          throw new IllegalArgumentException(String.format("Field '%s' is of unsupported type '%s'.",
-                                                           field.getName(), fieldSchema.getType()));
+        if (!type.isSimpleType() && !BigQueryUtil.SUPPORTED_COMPLEX_TYPES.contains(type)) {
+          throw new IllegalArgumentException(String.format("Field '%s' is of unsupported type '%s'.", name, type));
+        }
+
+        if (type == Schema.Type.ARRAY) {
+          Schema componentSchema = fieldSchema.getComponentSchema();
+
+          if (field.getSchema().isNullable()) {
+            throw new IllegalArgumentException(String.format("Field '%s' is of type array. " +
+                                                               "BigQuery does not allow nullable arrays.", name));
+          }
+
+          if (componentSchema.isNullable()) {
+            throw new IllegalArgumentException(String.format("Field '%s' has nullable array items, " +
```

Comment:
Field '%s' is an array of nullable values, but BigQuery does not allow null values in arrays